### ISSUE TITLE
OpenMPI 4.0.5 MPI_COMM_SPAWN child fails to MPI_INIT

### ISSUE THREAD

**Issue by nevion** [open-mpi/ompi#8189]
Successor to https://github.com/open-mpi/ompi/issues/4555

### What version of Open MPI are you using? (e.g., v3.0.5, v4.0.2, git branch name and hash, etc.)
4.0.5

### Describe how Open MPI was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)
Source build, from release tarball.

### Please describe the system on which you are running

* Operating system/version: Linux, Centos 8
* Computer hardware: x86_64
* Network type: ethernet, tcp

-----------------------------

## Details of the problem

This is mainly re-opening the issue already opened in https://github.com/open-mpi/ompi/issues/4555#issuecomment-721701091, which seems closed without resolution to the OP's post. 

To repeat original posting which didn't get ack'd by any of the typical devs - can you please test 4.0.5 on orte/test/mpi/simple_spawn.c - without any job manager - just ssh based host lists fail - I'm unable to find a working configuration with openmpi that doesn't freeze up in the listed sample in MPI_Init in the child and MPI_Comm_spawn in the parent:
```
mpirun -mca pml ob1 -mca btl tcp,self -H node1,node2 --map-by ppr:1:node:oversubscribe `pwd`/spawn_test
[207880193:0 pid 3698] starting up on node node1!
[207880193:1 pid 16532] starting up on node node2!
0 completed MPI_Init
Parent [pid 3698] about to spawn!
[207880194:2 pid 3701] starting up on node node1!
1 completed MPI_Init
Parent [pid 16532] about to spawn!
[207880194:1 pid 16535] starting up on node node2!
[207880194:2 pid 16536] starting up on node node2!
```
However, mpich-3.3.2 w/ nemesis worked on this example without issue on the same nodeset. I was even able to try it with '-map-by node -ppn 2' with mpich.

**Comment by rhc54** [comment]
I believe we closed it because there are no plans to go back and fix it on the 4.0 series. You might try the 4.1 series and see if it works there. I have verified that it works (as does singleton comm-spawn) on the master branch, which will soon be branched to v5.0.

**Comment by rhc54** [comment]
OMPI v5 is scheduled to release in the near future. In the interim, you are welcome to use a nightly tarball from the OMPI master or v5 branches (https://www.open-mpi.org/nightly/)
